### PR TITLE
fix(editChannel): fix videoQualityMode & add permissionOverwrites

### PR DESCRIPTION
**Pull request by aicushman**
Corrects the option key for "video_quality_mode" and adds option key "permission_overwrites" per the `PATCH /channels/{channel.id}` endpoint spec at https://discord.com/developers/docs/resources/channel#modify-channel

This commit does NOT fix the types associated with the modified function.

### CODE REVIEW

**Comment by DonovanDMC** [comment]
> This commit does NOT fix the types associated with the modified function.

the types are your responsibility when making a pull request

File: lib/Client.js [hunk 1]
```diff
@@ -1262,7 +1263,8 @@ class Client extends EventEmitter {
             rtc_region: options.rtcRegion,
             topic: options.topic,
             user_limit: options.userLimit,
-            voice_quality_mode: options.voiceQualityMode,
+            video_quality_mode: options.videoQualityMode,
+            permission_overwrites:options.permissionOverwrites,
```
Comment (DonovanDMC) [inline lib/Client.js]:
```suggestion
            permission_overwrites: options.permissionOverwrites,
```

**Comment by aicushman** [comment]
> > This commit does NOT fix the types associated with the modified function.
> 
> the types are your responsibility when making a pull request

Happy to, if I knew the drill for editing the index.d.ts file. 
My assumption was it'd be faster for a reviewer to do it than to explain it, but either works for me 👍

In any case, AFAIK all that needs to happen is
`- interface EditChannelOptions extends Omit<CreateChannelOptions, "permissionOverwrites" | "reason">`
`+ interface EditChannelOptions extends Omit<CreateChannelOptions, "reason">`

**Review by bsian03 — DISMISSED** [review]
Docs and object in alphabetical order please, also please do typings. Feel free to join the Discord server if you're confused about typings

**Review by DonovanDMC — APPROVED** [review]

**Pull request merged**